### PR TITLE
Update `Unknown error` to specific error message

### LINKED ISSUE #49713: redfish_facts results an Unknown error
##### SUMMARY

Hello `Ansible` community! 
When executing a simple [playbook](https://github.com/dnuka/automate/blob/master/ansible/redfish_101.yaml) with [redfish_facts](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/remote_management/redfish/redfish_facts.py) module, results `Unknown error`

<pre><font color="#DC322F">fatal: [HOST]: FAILED! =&gt; {</font>
<font color="#DC322F">    &quot;changed&quot;: false,</font>
<font color="#DC322F">    &quot;invocation&quot;: {</font>
<font color="#DC322F">        &quot;module_args&quot;: {</font>
<font color="#DC322F">            &quot;baseuri&quot;: &quot;http://127.0.0.1:8000/&quot;,</font>
<font color="#DC322F">            &quot;category&quot;: [</font>
<font color="#DC322F">                &quot;Manager&quot;</font>
<font color="#DC322F">            ],</font>
<font color="#DC322F">            &quot;command&quot;: [</font>
<font color="#DC322F">                &quot;all&quot;</font>
<font color="#DC322F">            ],</font>
<font color="#DC322F">            &quot;password&quot;: &quot;VALUE_SPECIFIED_IN_NO_LOG_PARAMETER&quot;,</font>
<font color="#DC322F">            &quot;username&quot;: &quot;foo&quot;</font>
<font color="#DC322F">        }</font>
<font color="#DC322F">    },</font>
<font color="#DC322F">    &quot;msg&quot;: &quot;Unknown error&quot;</font>
<font color="#DC322F">}</font>
</pre>

I'm unable to debug/improve my [playbook](https://github.com/dnuka/automate/blob/master/ansible/redfish_101.yaml) with an error like `Unknown error`. So, I think it would be great if we could improve this to much more verbose error message.

Thanks :)

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

redfish_facts

##### ANSIBLE VERSION

```paste below
ansible 2.7.4
python version = 3.7.1 (default, Oct 21 2018, 08:03:53) [GCC 8.2.0]
```

### PR DESCRIPTION
**Pull request by warusadura**
##### SUMMARY

Outputs the Exception received rather than just 'Unknown error'

before:
```
fatal: [HOST]: FAILED! => {"changed": false, "msg": "Unknown error"}
```

after:
```
fatal: [HOST]: FAILED! => {"changed": false, "msg": "Failed GET operation against Redfish API server: invalid literal for int() with base 10: ''"}
```

fixes: #49713 
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

redfish_facts

### CODE REVIEW

**Comment by warusadura** [comment]
I'm not sure why the CI tests failed. seems like a `timeout`. Is there a way I could re run the CI tests.

**Comment by mattclay** [comment]
I've restarted the failed CI jobs.

**Comment by warusadura** [comment]
@mattclay Thank you :)

**Comment by Shaps** [comment]
Not a `redfish_utils` module user, but the change looks ok to me

**Review by billdodd — APPROVED** [review]
Thanks for the update, @dnuka. Fix looks good.

shipit

File: lib/ansible/module_utils/redfish_utils.py [hunk 1]
```diff
@@ -35,8 +36,8 @@ def get_request(self, uri):
         except URLError as e:
             return {'ret': False, 'msg': "URL Error: %s" % e.reason}
         # Almost all errors should be caught above, but just in case
-        except Exception:
-            return {'ret': False, 'msg': "Unknown error"}
+        except Exception as e:
+            return {'ret': False, 'msg': to_text(e)}
```
Comment (bcoca) [inline lib/ansible/module_utils/redfish_utils.py]: the original was bad, this is slightly better, but a really good error message would include 'what i was trying to do':

`msg: 'Failed GET operation against Redfish API server: %s' % to_text(e)`

File: lib/ansible/module_utils/redfish_utils.py [hunk 2]
```diff
@@ -53,8 +54,8 @@ def post_request(self, uri, pyld, hdrs):
...
```
Comment (bcoca) [inline lib/ansible/module_utils/redfish_utils.py]: failed POST operation against ...

File: lib/ansible/module_utils/redfish_utils.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/module_utils/redfish_utils.py]: even better would be to show what specific operation we were attempting, but that does not seem clear w/o parsing the uri itself (for that, the caller should be processing the exception)

File: lib/ansible/module_utils/redfish_utils.py (hunk 2 above)
Comment (dnuka) [reply lib/ansible/module_utils/redfish_utils.py]: @bcoca Thank you for the review. Requested changes applied :)

**Comment by warusadura** [comment]
@bcoca  Thank you for the review. Requested changes applied :)

**Comment by warusadura** [comment]
@billdodd 
can you please recheck the new changes :)

**Review by billdodd — APPROVED** [review]
Updates look good. Thanks.

shipit

**Comment by jose-delarosa** [comment]
shipit

**Pull request merged**